game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/CombatValue.java

```diff
@@ -0,0 +1,98 @@
+package games.strategy.triplea.delegate.power.calculator;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.TerritoryEffect;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.delegate.Matches;
+import java.util.Collection;
+
+public interface CombatValue {
+
+  StrengthAndRollCalculator getRoll();
+
+  StrengthAndRollCalculator getStrength();
+
+  boolean isDefending();
+
+  GameData getGameData();
+
+  static CombatValue buildMain(
+      final Collection<Unit> allEnemyUnitsAliveOrWaitingToDie,
+      final Collection<Unit> allFriendlyUnitsAliveOrWaitingToDie,
+      final boolean defending,
+      final GameData data,
+      final Territory location,
+      final Collection<TerritoryEffect> territoryEffects) {
+
+    // Get all friendly supports
+    final AvailableSupports supportFromFriends =
+        AvailableSupports.getSortedSupport(
+            new SupportCalculator(
+                allFriendlyUnitsAliveOrWaitingToDie,
+                data.getUnitTypeList().getSupportRules(),
```

Comment:
If we are just using the support rules from the `GameData`, what do you think about passing those support rules directly to the method API? That should help create a more testable interface and drops a dependency on GameData.

Reply:
Yup, that would be part of the `Predicate` change you commented on earlier.  The `getSupportRules` and `getAaSupportRules` uses a similar `Predicate` inside of them and I'd like to unify the logic in some manner and simplify it.